scripts/docgen/snippets/transformFileText.ts

```diff
@@ -0,0 +1,44 @@
+import { SourceFile, SyntaxKind } from "ts-morph"
+import { PackageMetadata } from "../extract.js"
+import { SnippetTransformToggles } from "./index.js"
+
+export type ExtractFileSnippetContext = {
+    packageMetadata: PackageMetadata
+    transforms: SnippetTransformToggles
+}
+export const extractTransformText = (
```

Comment:
Let's call this `getTransformedText` since we are not necessarily extracting it if it didn't exist in the file originally.

This is also the first function that should actually use `project` directly since we are only using `ts-morph` for transformations.